### PR TITLE
feat: Lewis structures Style

### PR DESCRIPTION
**Pull request by wodeni**
# Description

This PR adds a Style file representing the Lewis dot structure of molecules, along with various examples. 

# Implementation strategy and design decisions

To declare atoms, I added all elements to the Domain and Style file using the following script:
```js
const chem = require("@chemistry/elements");

const elems = chem.ChemElements.getAll();
const styleBlocks = elems.map(
  (e) =>
    `
forall ${e.name} x {
  override x.symbol = "${e.symbol}"
  override x.id = ${e.id}
}
`
);

const domainDefs = elems.map(
  (e) =>
    `type ${e.name} <: Atom
`
);

// console.log(styleBlocks.join(""));
console.log(domainDefs.join(""));
```

In `lewis.style`, there are three layout stages:

* Adapt the existing `notTooClose` based layout for atoms and bonds (layout stage `general`)
* Attempt to make the diagram symmetric using @keenancrane's symmetric constraint (layout stage `symmetry`)
* Add electron pairs to atoms and encourage them to spread out and stay away from bonds (layout stage `electrons`)

# Examples with steps to reproduce them

# Checklist

- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new ESLint warnings
- [x] I have reviewed any generated registry diagram changes

# Open questions

N/A

### CODE REVIEW

File: packages/core/src/shapes/Shapes.ts [hunk 1]
```diff
@@ -277,7 +277,6 @@ const makeGroup = (
 const Group = ShapeDef({
   sampler: sampleGroup,
   constr: makeGroup,
-
```
Comment (samestep) [inline packages/core/src/shapes/Shapes.ts:280]: Seems like a random change 😛

File: packages/examples/src/molecules/water.substance [hunk 2]
```diff
@@ -0,0 +1,18 @@
+Hydrogen h
+Oxygen o1, o2
+Bond b1 := MakeSingleBond(h, o1)
+Bond b2 := MakeSingleBond(h, o2)
+-- Electron o11 := ValenceElectron(o1)
+-- Electron o12 := ValenceElectron(o1)
+-- Electron o13 := ValenceElectron(o1)
+-- Electron o14 := ValenceElectron(o1)
+-- Electron o15 := ValenceElectron(o1)
+-- Electron o16 := ValenceElectron(o1)
```
Comment (samestep) [inline packages/examples/src/molecules/water.substance:10]: Is there a reason for leaving these `ValenceElectron`s commented out in the Substance program?

File: packages/examples/src/molecules/lewis.style [hunk 3]
```diff
@@ -0,0 +1,849 @@
...
+  scalar symmetryWeight = 1
+  scalar symmetryDegree = 3
+  scalar bondAvoidWeight = 1
+}
+
+-- generated element list
+
+forall Hydrogen x {
+  override x.symbol = "H"
+  override x.id = 1
```
Comment (samestep) [inline packages/examples/src/molecules/lewis.style:20]: Why do we use `override` here?

**Review by samestep — APPROVED** [review]
Looks great to me! Just had a few questions.

File: packages/core/src/shapes/Shapes.ts (hunk 1 above)
Comment (wodeni) [reply packages/core/src/shapes/Shapes.ts:280]: yeah the blank line bothers me a bit

File: packages/examples/src/molecules/water.substance (hunk 2 above)
Comment (wodeni) [reply packages/examples/src/molecules/water.substance]: good catch! fixed

**Pull request merged**